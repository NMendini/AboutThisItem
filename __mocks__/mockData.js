module.exports = {
  details: {
    "highlights": [
        "MAGNI REPELLAT AUT: Facere odio fugiat nesciunt voluptatem rerum omnis accusantium aut adipisci.",
        "ATQUE CONSEQUATUR ET: Consequatur error fugit sint possimus.\nVel id cum dolor.",
        "IURE AT SOLUTA: Porro dolore libero nisi.\nVel rerum non aliquid ut.",
        "QUOS EUM VEL: Aspernatur excepturi qui eaque consequatur voluptatem id consectetur officia dicta.\nSit aut aliquid sunt dignissimos id est perferendis deleniti deleniti.\nFuga vel quo qui odit animi et id.\nQuos rerum corporis quis pariatur nulla sed id.\nUt consectetur temporibus.",
        "EAQUE VOLUPTATEM PARIATUR: Sit repellat rem veritatis nihil quaerat eum quia quis.",
        "ATQUE EA AUT: Corporis quas magnam adipisci consequatur dolore et ipsum et quae.\nSed hic omnis ducimus rerum odit possimus.\nTempore adipisci eos quod rem.\nEa quos architecto corporis.\nDolore in rem quisquam ut quia."
    ],
    "specifications": {
        "Item Number": "1",
        "Frozen": "aut dicta eligendi",
        "Rubber": "et odit voluptas",
        "Fresh": "quas qui iste",
        "Metal": "nostrum officia sunt",
        "Wooden": "dolor fugit nihil",
        "Granite": "enim architecto at",
        "Cotton": "sint repellendus velit",
        "Plastic": "omnis ratione iure",
        "Steel": "et quisquam animi",
        "Height": "17.25 inches",
        "Width": "24.07 inches",
        "Length": "27.23 inches",
        "Weight": "5.48 pounds"
    },
    "description": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support. Labore quasi praesentium reiciendis et expedita magnam ut. Aut ex quod voluptatem dolorem id et aut fuga et. Sit distinctio magni sit suscipit molestias quae et. Vel quo fugiat blanditiis placeat. Consequuntur itaque aliquam sequi ut.\n \rIusto quas corrupti tempora. Animi neque neque ut blanditiis aliquam eveniet rem. Sed officia reiciendis nulla vel aliquam eum iste tenetur.\n \rQuia debitis libero rem ad qui officia alias ut optio. Vel non qui iure. Nobis doloribus voluptatem quisquam omnis. Et rerum quae sint atque laborum tenetur excepturi ullam. Officiis et aut ea."
  },
  questions: [
    {
        "question_id": 1,
        "user_name": "Dedrick",
        "question": "Aut eum aut laboriosam id laborum vel omnis voluptas?",
        "created_at": "2020-04-02T17:51:57.000Z",
        "product_id": 91
    },
    {
        "question_id": 2,
        "user_name": "Freda",
        "question": "Ea quisquam atque molestiae non qui accusantium?",
        "created_at": "2020-05-18T20:14:35.000Z",
        "product_id": 91
    }
  ],
  answers: [
    {
        "answer_id": 1,
        "user_name": "Nichole",
        "question_id": 1,
        "answer": "Neque officiis quod quo perferendis voluptatum aut sed. Aut in pariatur tempora et animi saepe non. Iure neque quia qui qui qui excepturi. Quod vitae pariatur nisi neque quidem voluptates maxime qui.",
        "created_at": "2020-01-15T19:45:38.000Z",
        "helpful": 1,
        "not_helpful": 12,
        "product_id": 1
    },
    {
        "answer_id": 2,
        "user_name": "Arianna",
        "question_id": 1,
        "answer": "Qui sit vel aut molestiae. Expedita quia ipsam. Soluta accusantium aspernatur consequatur quibusdam magnam quas. Voluptas fuga sapiente nam distinctio deserunt fuga atque.",
        "created_at": "2020-05-03T22:34:22.000Z",
        "helpful": 19,
        "not_helpful": 27,
        "product_id": 1
    },
    {
      "answer_id": 3,
      "user_name": "Joshua",
      "question_id": 2,
      "answer": "Soluta optio deserunt ipsam aut hic nostrum quaerat. Voluptatem accusamus harum.",
      "created_at": "2020-03-09T08:11:04.000Z",
      "helpful": 16,
      "not_helpful": 35,
      "product_id": 1
    },
    {
        "answer_id": 4,
        "user_name": "Hubert",
        "question_id": 2,
        "answer": "Nisi distinctio praesentium. Qui maxime qui a laborum eum quia sed vero quae. Cupiditate voluptatem aut culpa dolores omnis enim voluptas recusandae. Assumenda et voluptatem sed unde impedit autem cupiditate cum.",
        "created_at": "2020-03-12T15:16:48.000Z",
        "helpful": 37,
        "not_helpful": 39,
        "product_id": 1
    }
  ]
}