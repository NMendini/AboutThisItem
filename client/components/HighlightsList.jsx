import React from 'react';
import HighlightsListEntry from './HighlightsListEntry.jsx';
import styles from '../assets/style.css';

const HighlightsList = function(props) {
  return (
    <ul className={styles.containerHighlights}>
      {
        props.highlights.map((item, index) =>
          <HighlightsListEntry highlight={item} key={index}/>
        )}
    </ul>
  );
};

export default HighlightsList;