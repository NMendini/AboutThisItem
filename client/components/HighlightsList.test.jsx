/**
 * @jest-environment jsdom
 */
import 'babel-polyfill';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import mockAxios from 'jest-mock-axios';
import { details, questions, answers }from '../../__mocks__/mockData.js';

// COMPONENTS
import HighlightsList from './HighlightsList.jsx';

let component;
let props;

beforeAll(async () => {
  props = {
    highlights: details.highlights
  }

  component = mount(<HighlightsList {...props}/>);


});

afterAll(async () => {
  // await component.unmount();
});

describe('HighlightsList Component', () => {
  it('should render HighlightsList component correctly', async () => {
      expect(component).toMatchSnapshot();
  });
});