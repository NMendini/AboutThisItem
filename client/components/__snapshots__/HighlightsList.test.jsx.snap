// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightsList Component should render HighlightsList component correctly 1`] = `
<HighlightsList
  highlights={
    Array [
      "MAGNI REPELLAT AUT: Facere odio fugiat nesciunt voluptatem rerum omnis accusantium aut adipisci.",
      "ATQUE CONSEQUATUR ET: Consequatur error fugit sint possimus.
Vel id cum dolor.",
      "IURE AT SOLUTA: Porro dolore libero nisi.
Vel rerum non aliquid ut.",
      "QUOS EUM VEL: Aspernatur excepturi qui eaque consequatur voluptatem id consectetur officia dicta.
Sit aut aliquid sunt dignissimos id est perferendis deleniti deleniti.
Fuga vel quo qui odit animi et id.
Quos rerum corporis quis pariatur nulla sed id.
Ut consectetur temporibus.",
      "EAQUE VOLUPTATEM PARIATUR: Sit repellat rem veritatis nihil quaerat eum quia quis.",
      "ATQUE EA AUT: Corporis quas magnam adipisci consequatur dolore et ipsum et quae.
Sed hic omnis ducimus rerum odit possimus.
Tempore adipisci eos quod rem.
Ea quos architecto corporis.
Dolore in rem quisquam ut quia.",
    ]
  }
>
  <ul
    className="containerHighlights"
  >
    <HighlightsListEntry
      highlight="MAGNI REPELLAT AUT: Facere odio fugiat nesciunt voluptatem rerum omnis accusantium aut adipisci."
      key="0"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            MAGNI REPELLAT AUT: Facere odio fugiat nesciunt voluptatem rerum omnis accusantium aut adipisci.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
    <HighlightsListEntry
      highlight="ATQUE CONSEQUATUR ET: Consequatur error fugit sint possimus.
Vel id cum dolor."
      key="1"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            ATQUE CONSEQUATUR ET: Consequatur error fugit sint possimus.
Vel id cum dolor.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
    <HighlightsListEntry
      highlight="IURE AT SOLUTA: Porro dolore libero nisi.
Vel rerum non aliquid ut."
      key="2"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            IURE AT SOLUTA: Porro dolore libero nisi.
Vel rerum non aliquid ut.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
    <HighlightsListEntry
      highlight="QUOS EUM VEL: Aspernatur excepturi qui eaque consequatur voluptatem id consectetur officia dicta.
Sit aut aliquid sunt dignissimos id est perferendis deleniti deleniti.
Fuga vel quo qui odit animi et id.
Quos rerum corporis quis pariatur nulla sed id.
Ut consectetur temporibus."
      key="3"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            QUOS EUM VEL: Aspernatur excepturi qui eaque consequatur voluptatem id consectetur officia dicta.
Sit aut aliquid sunt dignissimos id est perferendis deleniti deleniti.
Fuga vel quo qui odit animi et id.
Quos rerum corporis quis pariatur nulla sed id.
Ut consectetur temporibus.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
    <HighlightsListEntry
      highlight="EAQUE VOLUPTATEM PARIATUR: Sit repellat rem veritatis nihil quaerat eum quia quis."
      key="4"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            EAQUE VOLUPTATEM PARIATUR: Sit repellat rem veritatis nihil quaerat eum quia quis.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
    <HighlightsListEntry
      highlight="ATQUE EA AUT: Corporis quas magnam adipisci consequatur dolore et ipsum et quae.
Sed hic omnis ducimus rerum odit possimus.
Tempore adipisci eos quod rem.
Ea quos architecto corporis.
Dolore in rem quisquam ut quia."
      key="5"
    >
      <div
        className="highlightElement"
      >
        <li
          className="highlightItem"
        >
          <span>
            ATQUE EA AUT: Corporis quas magnam adipisci consequatur dolore et ipsum et quae.
Sed hic omnis ducimus rerum odit possimus.
Tempore adipisci eos quod rem.
Ea quos architecto corporis.
Dolore in rem quisquam ut quia.
          </span>
        </li>
      </div>
    </HighlightsListEntry>
  </ul>
</HighlightsList>
`;
